integration-libs/digital-payments/src/checkout/adapters/occ-digital-payments.adapter.ts

```diff
@@ -0,0 +1,48 @@
+import { CURRENT_USER, CURRENT_CART } from '../../utils/dp-constants';
+import { HttpClient, HttpParams } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import {
+  OccEndpointsService,
+  Occ,
+  HttpParamsURIEncoder,
+} from '@spartacus/core';
+import { Observable } from 'rxjs';
+import { DigitalPaymentsAdapter } from './digital-payments.adapter';
+import { DpPaymentRequest } from '../models';
+
+@Injectable()
+export class OccDigitalPaymentsAdapter implements DigitalPaymentsAdapter {
+  private readonly paramEncoder = new HttpParamsURIEncoder();
+
+  constructor(
+    private http: HttpClient,
+    private occEndpoints: OccEndpointsService
+  ) {}
+
+  createPaymentRequest(
+    userId = CURRENT_USER,
+    cartId = CURRENT_CART
+  ): Observable<DpPaymentRequest> {
+    return this.http.post<any>(
+      `${this.occEndpoints.getBaseUrl()}/users/${userId}/carts/${cartId}/payment/digitalPayments/request`,
```

Comment:
Instead of hardcoding url here it should be defined in `OccEndpoints` and it should be fetched with `buildUrl` passing onlu the required params (userId, cartId).

Applies to second method as well.

Reply:
changes are done.